protocol/synthetix/contracts/modules/core/MarketManagerModule.sol

```diff
@@ -329,38 +328,101 @@ contract MarketManagerModule is IMarketManagerModule {
     /**
      * @inheritdoc IMarketManagerModule
      */
-    function setMarketMinDelegateTime(uint128 marketId, uint32 minDelegateTime) external override {
+    function setUndelegateCollateralDelay(
+        uint128 marketId,
+        uint32 undelegateCollateralDelay
+    ) external override {
         Market.Data storage market = Market.load(marketId);
 
         if (ERC2771Context._msgSender() != market.marketAddress)
             revert AccessError.Unauthorized(ERC2771Context._msgSender());
 
-        // min delegate time should not be unreasonably long
-        uint256 maxMinDelegateTime = Config.readUint(
-            _CONFIG_SET_MARKET_MIN_DELEGATE_MAX,
-            86400 * 30
-        );
+        market.undelegateCollateralDelay = undelegateCollateralDelay;
 
-        if (minDelegateTime > maxMinDelegateTime) {
-            revert ParameterError.InvalidParameter("minDelegateTime", "must not be too large");
-        }
+        emit SetUndelegateCollateralDelay(marketId, undelegateCollateralDelay);
+    }
+
+    /**
+     * @inheritdoc IMarketManagerModule
+     */
+    function getUndelegateCollateralDelay(
+        uint128 marketId
+    ) external view override returns (uint32) {
+        return Market.load(marketId).undelegateCollateralDelay;
+    }
+
+    /**
+     * @inheritdoc IMarketManagerModule
+     */
+    function setUndelegateCollateralWindow(
```

Comment:
please combine these individual `setUndelegateCollateral...` functions into a single setter that takes a tuple/structure.

its likely that such values will be always set/changed at the same time, reduces surface area, and 

same goes for getter.